tensorly/contrib/decomposition/tt_TTOI.py

```diff
@@ -0,0 +1,260 @@
+            factors_list_tmp.append(tl.tensor(left_residuals[n_dim - 2], **context))
+            full_tensor_tmp = tl.tensor(tt_to_tensor(factors_list_tmp), **context)
+            if return_errors:
+                error_list.append(tl.norm(full_tensor_tmp - data_tensor, 2))
+            if trajectory:
+                factors.append(factors_list_tmp)
+                full_tensor.append(full_tensor_tmp)
+            elif iteration == np.ceil(n_iter/2) - 1 and np.ceil(n_iter / 2) * 2 > n_iter:
+                factors = factors_list_tmp
+                full_tensor = full_tensor_tmp
+        
+        if np.ceil(n_iter / 2) * 2 == n_iter:
+            # perform backward update
+            # initialize right_singular_vectors: right_singular_vectors will be a list of estimated right singular spaces at the current or previous iteration
+            right_singular_vectors = []
+            _, _, V_tmp = tl.tenalg.svd_interface(
+                    matrix=tl.reshape(
+                        left_residuals[n_dim - 2], (rank[n_dim - 1], shape[n_dim - 1])
+                        ),
+                    n_eigenvecs=rank[n_dim - 1],
+                    )
+            V_tmp = tl.transpose(V_tmp)
+            right_singular_vectors.append(
+                tl.reshape(V_tmp, (rank[n_dim], shape[n_dim - 1], rank[n_dim - 1]))
+                )
+            # estimate the 2nd to (d-1)th right singular spaces
+            for mode in range(n_dim - 2):
+                # compress left_residuals from the right
+                R_tmp_r = sequential_prod(
+                    left_residuals[n_dim - mode - 3],
+                    right_singular_vectors[0 : (mode + 1)],
+                    "right",
+                    )
+                _, _, V_tmp = tl.tenalg.svd_interface(
+                    matrix=tl.reshape(
+                        R_tmp_r,
+                        (
+                            rank[n_dim - mode - 2],
+                            shape[n_dim - mode - 2] * rank[n_dim - mode - 1],
+                            ),
+                        ),
+                    n_eigenvecs=rank[n_dim - mode - 2],
+                    )
+                right_singular_vectors.append(
+                    tl.transpose(
+                        tl.reshape(
+                            V_tmp,
+                            (
+                                rank[n_dim - mode - 2],
+                                shape[n_dim - mode - 2],
+                                rank[n_dim - mode - 1],
+                                ),
+                            )
+                        )
+                    )
+            Residual_right = sequential_prod(
+                data_tensor_extended, right_singular_vectors, left_to_right=False
+                )
+            if trajectory or return_errors or iteration == np.ceil(n_iter/2) - 1:
+                factors_list_tmp = []
+                factors_list_tmp.append(tl.tensor(Residual_right, **context))
```

Comment:
Same

Reply:
> I left a few comments. There are some`tl.tensor`s which I think are not necessary. Removing them could improve code readability.

Thanks for the suggestions. I have removed them.